x/emissions/keeper/queryserver/query_server_inferences.go

```diff
@@ -167,17 +187,103 @@ func (qs queryServer) GetLatestAvailableNetworkInference(
 			inferenceBlockHeight,
 			previousLossBlockHeight,
 		)
-
 	if err != nil {
 		return nil, err
 	}
 
+	ciRawPercentiles, ciValues, err :=
+		qs.GetConfidenceIntervalsForInferenceData(
+			networkInferences,
+			forecastImpliedInferenceByWorker,
+			infererWeights,
+			forecasterWeights,
+		)
+
+	if ciRawPercentiles == nil {
+		ciRawPercentiles = []alloraMath.Dec{}
+	}
+
+	if ciValues == nil {
+		ciValues = []alloraMath.Dec{}
+	}
+
 	return &types.QueryLatestNetworkInferencesResponse{
-		NetworkInferences:         networkInferences,
-		InfererWeights:            synth.ConvertWeightsToArrays(infererWeights),
-		ForecasterWeights:         synth.ConvertWeightsToArrays(forecasterWeights),
-		ForecastImpliedInferences: synth.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
-		InferenceBlockHeight:      inferenceBlockHeight,
-		LossBlockHeight:           previousLossBlockHeight,
+		NetworkInferences:                networkInferences,
+		InfererWeights:                   synth.ConvertWeightsToArrays(infererWeights),
+		ForecasterWeights:                synth.ConvertWeightsToArrays(forecasterWeights),
+		ForecastImpliedInferences:        synth.ConvertForecastImpliedInferencesToArrays(forecastImpliedInferenceByWorker),
+		InferenceBlockHeight:             inferenceBlockHeight,
+		LossBlockHeight:                  previousLossBlockHeight,
+		ConfidenceIntervalRawPercentiles: ciRawPercentiles,
+		ConfidenceIntervalValues:         ciValues,
 	}, nil
 }
+
+func (qs queryServer) GetConfidenceIntervalsForInferenceData(
+	networkInferences *emissions.ValueBundle,
+	forecastImpliedInferenceByWorker map[string]*emissions.Inference,
+	infererWeights map[string]alloraMath.Dec,
+	forecasterWeights map[string]alloraMath.Dec,
+) ([]alloraMath.Dec, []alloraMath.Dec, error) {
+
+	inferersToInference := make(map[string]alloraMath.Dec)
+	for _, infererValue := range networkInferences.InfererValues {
+		inferersToInference[infererValue.Worker] = infererValue.Value
+	}
+
+	forecastersToInference := make(map[string]alloraMath.Dec)
+	for _, forecasterValue := range networkInferences.ForecasterValues {
+		forecastersToInference[forecasterValue.Worker] = forecasterValue.Value
+	}
```

Comment:
I don't understand why these maps exist at all... If anything they make the order of elements in the query nondeterministic, which possibly has effects on downstream applications

Reply:
proposed in my latest commit